### PR TITLE
Workaround k8s exec probe timeoutSeconds

### PR DESCRIPTION
**Pull request by ghost**
#### What this PR does / why we need it:

We've found that redis-cli can hang and not return in some
circumstances (e.g. when the persistentVolume is not responding).

Kubernetes probes that run in Docker containers currently ignore
the timeoutSeconds specified in the probe. [0]

To workaround this, pass the timeoutSeconds as a parameter to the
scripts, and use the timeout tool to send a SIGKILL to redis-cli
if it's hasn't responded after the timeout expires.

[0] https://github.com/kubernetes/kubernetes/pull/58925

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

**Review by juan131 — APPROVED** [review]
Thanks for this workaround! Very nice!

**Comment by juan131** [comment]
Could you please update the version in the `Chart.yaml`. There was a new version of the chart and there're some conflicts.

**Comment by stsui-ctech** [comment]
Thanks for the feedback. Conflict's resolved now.

File: stable/redis/templates/_helpers.tpl [hunk 1]
```diff
@@ -103,6 +103,7 @@ readinessProbe:
     - sh
     - -c
     - /health/ping_local_and_master.sh
+    - {{ $readinessProbe.timeoutSeconds | default .Values.master.readinessProbe.timeoutSeconds }}
```
Comment (juan131) [inline stable/redis/templates/_helpers.tpl]: The tests are failing, could you adapt the command to use so it looks like the one below:

```
         - bash
         - -ec
         - |
            /health/ping_local_and_master.sh
            {{ $livenessProbe.timeoutSeconds | default .Values.master.livenessProbe.timeoutSeconds }}
```

I think that should fix the tests.

**Review by juan131 — CHANGES_REQUESTED** [review]

File: stable/redis/templates/_helpers.tpl (hunk 1 above)
Comment (ghost) [reply stable/redis/templates/_helpers.tpl]: I changed it the same way I changed the statefulset, let me know if you prefer both be the way you suggested instead.

Comment (juan131) [reply stable/redis/templates/_helpers.tpl]: It's okay like you did. Thanks!

**Review by juan131 — APPROVED** [review]

**Pull request merged**